ci/tests/backend/alba.py

```diff
@@ -232,17 +238,83 @@ def is_bucket_count_valid_with_policy(bucket_count, policies):
         policy = tuple(policy)
         for entry in bucket_count:
             bc_policy = entry[0]
-            print policy
             pol_k, pol_m, pol_c, pol_x = tuple(policy)
-            print bc_policy, tuple(bc_policy)
             bc_k, bc_m, bc_c, bc_x = tuple(bc_policy)
             safe = (pol_k == bc_k) and (pol_m == bc_m) and (bc_c >= pol_c) and (bc_x <= pol_c)
 
     return safe
 
 
-def claim_disks(alba_backend, nr_of_disks, disk_type='sata'):
-    pass
+def filter_disks(available_disks, amount, disk_type=''):
+    hdds, ssds = get_physical_disks(GRID_IP)
+    count = 0
+    filtered_disks = list()
+
+    for disk_name in available_disks:
+        if disk_type == 'SATA':
+            list_to_check = hdds.itervalues()
+        elif disk_type == 'SSD':
+            list_to_check = ssds.itervalues()
+        else:
+            hdds.update(ssds)
+            list_to_check = hdds.itervalues()
+
+        for disk in list_to_check:
+            if disk_name == disk['name']:
+                filtered_disks.append(disk['name'])
+                count += 1
+        if count == amount:
+            break
+    return filtered_disks
+
+
+def initialise_disks(alba_backend, nr_of_disks, disk_type=''):
+    # assume no disks are claimed by a remote environment
+    api = Connection.get_connection()
+    all_disks = api.fetch('alba/backends', alba_backend['guid'])['all_disks']
+    alba_node = AlbaNodeList.get_albanode_by_ip(GRID_IP)
+
+    initialised_disks = [disk['name'] for disk in all_disks if disk['status'] == 'available']
+    nr_of_disks_to_init = nr_of_disks - len(initialised_disks)
+    if nr_of_disks_to_init <= 0:
+        return True
+
+    initializable_disks = [disk['name'] for disk in all_disks if disk['status'] in 'uninitialized']
```

Comment:
Hmm....
'initialized' in 'uninitialized' --> True

if alba or whoever decides to return 'initialized' for status instead of available, we kinda have a problem here